heron/api/src/java/org/apache/heron/streamlet/impl/BuilderImpl.java

```diff
@@ -44,6 +46,7 @@ public BuilderImpl() {
 
   @Override
   public <R> Streamlet<R> newSource(SerializableSupplier<R> supplier) {
+    Preconditions.checkNotNull(supplier, "supplier must not be null.");
```

Comment:
why use Preconditions? require() is not enough?

We are hoping to get rid of guava library since it is too big. :(

Reply:
Yes, `require` can be used by moving from `StreamletImpl` to `StreamletUtils` as the current patch